File: bindings/dart/example/bithumb_provider.dart
import 'package:maxt/maxt.dart';

Future<void> main() async {
  await Maxt.initialize();
  try {
    final adapter = BithumbAdapter();
    final warnings = await adapter.marketWarnings();
    final notices = await adapter.notices(5);
    final fees = await adapter.transferFees('BTC');
    print(
      '${warnings.length} warning rows, ${notices.length} notices, ${fees.length} BTC fee rows',
    );
  } finally {
    await Maxt.dispose();
  }
}

File: bindings/typescript/examples/bithumb-provider.mjs
// Read public Bithumb market warnings, notices, and transfer fees.
import { BithumbAdapter, initialize } from "@jabdori/maxt/node";

await initialize();

const adapter = new BithumbAdapter();
const [warnings, notices, fees] = await Promise.all([
  adapter.marketWarnings(),
  adapter.notices(5),
  adapter.transferFees("BTC"),
]);
console.log(`${warnings.length} warning rows, ${notices.length} notices, ${fees.length} BTC fee rows`);
